piet-cairo/src/text.rs

```diff
@@ -166,7 +167,12 @@ impl Text for CairoText {
     type TextLayoutBuilder = CairoTextLayoutBuilder;
 
     fn font_family(&mut self, family_name: &str) -> Option<FontFamily> {
-        //TODO: Veryify that a family exists with the requested name
+        // The pango documentation says this is always a stirng, and never null. I trust them on that.
+        // Would be weird to have a font family without a name anyways.
+        self.pango_context
+            .list_families()
+            .iter()
+            .find(|family| family.name().unwrap().as_str() == family_name);
```

Comment:
As a note: a fancier version of this might do things like return `Helvetica Neue` for `Helvetica` if `Helvetica` isn't present, and it might also want to be case-insensitive?


Additionally, it looks like we aren't using this check? If we don't find the font we should return `None`.

Reply:
> Additionally, it looks like we aren't using this check?

What check do you mean? I can add the `Helvetica` -> `Helvetica Neue` check if thats what you mean.

Reply:
What I mean is that we're looking at the families to see if `family_name` exists, but we're still always returning `Some` from this function.

We may also want to keep around a `HashSet` of family names so that we aren't doing a linear search each time, but that can be future work.

Reply:
Just for fun here are some useful guidelines on font matching: https://drafts.csswg.org/css-fonts-3/#font-matching-algorithm

Reply:
> What I mean is that we're looking at the families to see if family_name exists, but we're still always returning Some from this function.

I feel really stupid now. I just noticed this as well while looking at the code again.

As for the matching, I am honestly not sure how I would go about implementing that. Stripping away all the stuff we dont have/need, im left with the only restriction being "a case insensitive match". it should be easy to match case insensitively, but that doesnt solve the `Helvetica` -> `Helvetica Neue` match you wanted. I could check for substrings, I think thats easiest. It wont match the other way around, but we could check vise-versa as well?

Reply:
There exists a function in pango for matching, I just use that.

Reply:
Is this a change you're going to make, or is it in the current version? I didn't see it.

Reply:
I currently made a change so that it uses the pango matching function.

Reply:
(the `better_match` function)